totalRP3/modules/register/characters/PlayerMapScan/PlayerMapScanner.lua

```diff
@@ -0,0 +1,126 @@
+---@type AddOn_TotalRP3
+local AddOn_TotalRP3 = AddOn_TotalRP3;
+
+--region Lua imports
+local tonumber = tonumber;
+local insert = table.insert;
+--endregion
+
+--region WoW imports
+local CreateVector2D = CreateVector2D;
+--endregion
+
+--region Total RP 3 imports
+local loc = TRP3_API.loc;
+local broadcast = AddOn_TotalRP3.Communications.broadcast;
+local registerConfigKey = TRP3_API.configuration.registerConfigKey;
+local getConfigValue = TRP3_API.configuration.getValue;
+local get = TRP3_API.profile.getData;
+local Map = AddOn_TotalRP3.Map;
+--endregion
+
+local CONFIG_ENABLE_MAP_LOCATION = "register_map_location";
+local CONFIG_DISABLE_MAP_LOCATION_ON_OOC = "register_map_location_ooc";
+
+local function shouldAnswerToLocationRequest()
+	return getConfigValue(CONFIG_ENABLE_MAP_LOCATION)
+		and (not getConfigValue(CONFIG_DISABLE_MAP_LOCATION_ON_OOC) or get("player/character/RP") ~= 2)
+end
+
+TRP3_API.Events.registerCallback(TRP3_API.Events.WORKFLOW_ON_LOADED, function()
+
+	registerConfigKey(CONFIG_ENABLE_MAP_LOCATION, true);
+	registerConfigKey(CONFIG_DISABLE_MAP_LOCATION_ON_OOC, false);
+
+	insert(TRP3_API.register.CONFIG_STRUCTURE.elements, {
+		inherit = "TRP3_ConfigH1",
+		title = loc.CO_LOCATION,
+	});
+	insert(TRP3_API.register.CONFIG_STRUCTURE.elements, {
+		inherit = "TRP3_ConfigCheck",
+		title = loc.CO_LOCATION_ACTIVATE,
+		help = loc.CO_LOCATION_ACTIVATE_TT,
+		configKey = CONFIG_ENABLE_MAP_LOCATION,
+	});
+	insert(TRP3_API.register.CONFIG_STRUCTURE.elements, {
+		inherit = "TRP3_ConfigCheck",
+		title = loc.CO_LOCATION_DISABLE_OOC,
+		help = loc.CO_LOCATION_DISABLE_OOC_TT,
+		configKey = CONFIG_DISABLE_MAP_LOCATION_ON_OOC,
+		dependentOnOptions = { CONFIG_ENABLE_MAP_LOCATION },
+	});
+end)
+
+local SCAN_COMMAND = "C_SCAN";
+---@type MapScanner
+local playerMapScanner = AddOn_TotalRP3.MapScanner("playerScan")
+-- Set scan display properties
+playerMapScanner.scanIcon = "Achievement_GuildPerk_EverybodysFriend"
+playerMapScanner.scanOptionText = loc.MAP_SCAN_CHAR;
+playerMapScanner.scanTitle = loc.MAP_SCAN_CHAR_TITLE;
+-- Indicate the name of the pin template to use with this scan.
+-- The MapDataProvider will use this template to generate the pin
+playerMapScanner.dataProviderTemplate = TRP3_PlayerMapPinMixin.TEMPLATE_NAME;
+
+--region Scan behavior
+function playerMapScanner:Scan()
+	broadcast.broadcast(SCAN_COMMAND, Map.getDisplayedMapID());
+end
+
+-- Players can only scan for other players in zones where it is possible to retrieve player coordinates.
+function playerMapScanner:CanScan()
+	if getConfigValue(CONFIG_ENABLE_MAP_LOCATION) then
+		local x, y = Map.getPlayerCoordinates()
+		if x and y then
+			return true;
+		end
+	end
+	return false;
+end
+--endregion
+
+--region Broadcast commands
+broadcast.registerCommand(SCAN_COMMAND, function(sender, mapID)
+	if Map.playerCanSeeTarget(sender) then
+		mapID = tonumber(mapID);
+		if shouldAnswerToLocationRequest() and Map.playerCanSeeTarget(sender) then
```

Comment:
That's the second time you test `Map.playerCanSeeTarget(sender)`

Reply:
Yes. We check on the sender and receiver side, just to be extra sure. Checking before sending avoid answering uselessly, and checking when receiving is to make sure we don't have invalid data received.

Reply:
Wait no, I'm dumb. Sorry.